solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetRecordDatesFilter.java

```diff
@@ -16,33 +16,30 @@
 package edu.unc.lib.dl.data.ingest.solr.filter;
 
 import java.text.ParseException;
-
 import edu.unc.lib.dl.data.ingest.solr.exception.IndexingException;
 import edu.unc.lib.dl.data.ingest.solr.indexing.DocumentIndexingPackage;
-import edu.unc.lib.dl.util.ContentModelHelper.FedoraProperty;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.rdf.Fcrepo4Repository;
 
 /**
- * Indexing filter which extracts Fedora generated dates about the creation and modification state of the object
+ * Indexing filter that extracts Fedora-generated dates about the creation and modification state of an object
  * being indexed.
  * 
  * Sets: dateAdded, dateUpdated
- * @author bbpennel
+ * @author bbpennel, harring
  *
  */
-public class SetRecordDatesFilter extends AbstractIndexDocumentFilter {
+public class SetRecordDatesFilter {
 	
-	@Override
 	public void filter(DocumentIndexingPackage dip) throws IndexingException {
+		ContentObject obj = dip.getContentObject();
+		String dateAdded = obj.getResource().getPropertyResourceValue(Fcrepo4Repository.created).toString();
```

Comment:
In case you're interested, I had to do something like this in my fcrepo work.  You can get a date back "directly" from the graph since the value is typed.

```
XSDDateTime created = (XSDDateTime) obj.getResource()
                            .getProperty(CREATED_DATE).getLiteral().getValue();
Date createdDate = created.asCalendar().getTime()
```

This might save you the parsing exception catch down below, otherwise it doesn't make much difference.